lib/docker/imageservice.go

```diff
@@ -192,7 +226,19 @@ func (r *imageService) Sync(ctx context.Context, dir string, progress utils.Prin
 		if err != nil {
 			return nil, trace.Wrap(err)
 		}
-		remoteRepo, err := r.remoteStore.Repository(ctx, localRepoName)
+		remoteRepoName := localRepoName
+		// If the registry prefix was specified, rewrite all images to point
+		// to that domain - some registries (such as OpenShift) require a
+		// specific namespace where images should be pushed so an image like
+		// gravitational/debian-tall would become <namespace>/debian-tall.
```

Comment:
I wonder what the potential for conflicts here would be. Ie gravitational/debian-tall and quay.io/debian-tell overwrite eachother into <namespace>/debian-tall?? I don't see it in the online registries, but I think distribution supports multiple levels, so would something like gravitational/debian-tall to <namespace>/gravitational/debian-tall work?

Reply:
Yeah, they will overwrite each other. I tried it with multiple levels and it doesn't work unfortunately. At least the OpenShift registry requires you to have just `<namespace>/debian-tall`.